Lib/test/test_pyclbr.py

```diff
@@ -150,6 +154,121 @@ def test_decorators(self):
         #
         self.checkModule('test.pyclbr_input', ignore=['om'])
 
+    def test_nested(self):
+
+        def clbr_from_tuple(t, store, parent=None, lineno=1):
+            '''Create pyclbr objects from the given tuple t.'''
+            name = t[0]
+            obj = pickp(name)
+            if parent is not None:
+                store = store[parent].objects
+            ob_name = name.split()[1]
+            store[ob_name] = obj(name=ob_name, lineno=lineno, parent=parent)
+            parent = ob_name
+
+            for item in t[1:]:
+                lineno += 1
+                if isinstance(item, str):
+                    obj = pickp(item)
+                    ob_name = item.split()[1]
+                    store[parent].objects[ob_name] = obj(
+                            name=ob_name, lineno=lineno, parent=parent)
+                else:
+                    lineno = clbr_from_tuple(item, store, parent, lineno)
+
+            return lineno
+
+        def tuple_to_py(t, output, indent=0):
+            '''Write python code to output according to the given tuple.'''
+            name = t[0]
+            output.write('{}{}():'.format(' ' * indent, name))
+            indent += 2
+
+            if not t[1:]:
+                output.write(' pass')
+            output.write('\n')
+
+            for item in t[1:]:
+                if isinstance(item, str):
+                    output.write('{}{}(): pass\n'.format(' ' * indent, item))
+                else:
+                    tuple_to_py(item, output, indent)
+
+        # Nested "thing" to test.
+        sample = (
+                ("class A",
+                    ("class B",
+                        ("def a",
+                            "def b")),
+                    "def c"),
+                ("def d",
+                    ("def e",
+                        ("class C",
+                            "def f")
+                     ),
+                 "def g")
+                )
+
+        pclass = partial(pyclbr.Class, module=None, file=None, super=None)
+        pfunc = partial(pyclbr.Function, module=None, file=None)
+
+        def pickp(name):
+            return pclass if name.startswith('class') else pfunc
+
+        # Create a module for storing the Python code.
```

Comment:
Split done.  Test not yet simplified because _create_tree needs 5 arguments, not 2.  I would still like to simplify by new tests, but since they are working, this could wait for a separate pyclbr tests issue.